core/src/main/java/io/questdb/griffin/HttpSqlExecutionInterruptor.java

```diff
@@ -0,0 +1,66 @@
+package io.questdb.griffin;
+
+import java.io.Closeable;
+
+import io.questdb.cairo.CairoException;
+import io.questdb.network.NetworkFacade;
+import io.questdb.std.Unsafe;
+
+public class HttpSqlExecutionInterruptor implements SqlExecutionInterruptor, Closeable {
+    private final NetworkFacade nf;
+    private final int nIterrationsPerCheck;
+    private final int bufferSize;
+    private long buffer;
+    private int nIterrationsSinceCheck;
+    private long fd;
+
+    public HttpSqlExecutionInterruptor(NetworkFacade nf, int nIterrationsPerCheck, int bufferSize) {
+        super();
+        this.nf = nf;
+        this.nIterrationsPerCheck = nIterrationsPerCheck;
+        this.bufferSize = bufferSize;
+        buffer = Unsafe.malloc(bufferSize);
+    }
+
+    @Override
+    public void checkInterrupted() {
+        if (nIterrationsSinceCheck == nIterrationsPerCheck) {
```

Comment:
this method is or can be on a "hot" data path. We should pay special attention to branch prediction here and inlining. Perhaps microbench of this method's impact would be good to have.

Without microbech results I would suggest this method styles as follows:

```java
if (++nIterrationsSinceCheck < nIterrationsPerCheck) {
    return;
}
doHeavyCheck();